libs/application-generic/src/usecases/upsert-preferences/upsert-preferences.command.ts

```diff
@@ -1,17 +1,108 @@
-import { IsDefined, IsEnum } from 'class-validator';
-import { PreferencesTypeEnum, WorkflowPreferencesPartial } from '@novu/shared';
+import {
```

Comment:
The changes here are purely to strengthen validation for the `UpsertPreferences` use-case, which is used widely across the API for Workflow syncing, User preference updates (via Dashboard), and both Subscriber Global & Workflow updates via Inbox.